script.js

```diff
@@ -30,3 +81,149 @@ menuLinks.forEach((link) => {
+  popBtns.setAttribute('class', 'pop-btn');
+  const liveBtn = document.createElement('button');
+  liveBtn.setAttribute('class', 'live-btn');
+  liveBtn.setAttribute('type', 'button');
+  liveBtn.innerHTML = 'See live <i class="fas fa-arrow-circle-right fa-lg"></i>';
+  const sourceBtn = document.createElement('button');
+  sourceBtn.setAttribute('class', 'source-btn');
+  sourceBtn.setAttribute('type', 'button');
+  sourceBtn.innerHTML = 'See Source <i class="fab fa-github fa-lg"></i>';
+
+  const closeBtnPop = document.createElement('i');
+  closeBtnPop.className = 'fas fa-times';
+  closeBtnPop.addEventListener('click', () => {
+    const popup = document.querySelector('.project-pop');
+    if (popup) {
+      popup.parentElement.removeChild(popup);
+    }
+  });
+
+  techPop.append(techPopType);
+  techPop.append(techPopSpan1);
+  techPop.append(techPopDevType);
+  techPop.append(techPopSpan2);
+  techPop.append(techPopDevYear);
+  ProjectPopImageDiv.append(ProjectPopImage);
+  usedLang.append(usedLangP1);
+  usedLang.append(usedLangP2);
+  usedLang.append(usedLangP3);
+  popBtns.append(liveBtn);
+  popBtns.append(sourceBtn);
+  container10.append(usedLang);
+  container10.append(span);
+  container10.append(popBtns);
+  projectPopContainer.append(closeBtnPop);
+  projectPopContainer.append(projectPopName);
+  projectPopContainer.append(techPop);
+  projectPopContainer.append(ProjectPopImageDiv);
+  projectPopContainer.append(ProjectPopDesc);
+  projectPopContainer.append(container10);
+  projectPop.append(projectPopContainer);
+  return projectPop;
+}
+const { body } = document;
+
+document.querySelectorAll('.pop-btn').forEach((btn) => {
+  btn.addEventListener('click', (event) => {
+    if (!document.querySelector('.project-pop')) {
+      const popupData = data[parseInt(event.target.id, 10)];
+      const popup = CreatePopup(popupData);
+      body.append(popup);
+    }
+  });
+});
+
+// ######## form client side validation ########
+const form = document.getElementById('form');
+const email = document.getElementById('email');
+const errorMsg = document.getElementById('error-msg');
+
+function setErrorFor(input, message) {
+  const formControl = input.parentElement;
+  formControl.className = 'form-control error';
+  errorMsg.innerHTML = message;
+  errorMsg.style.display = 'block';
+}
+
+function setSuccessFor(input) {
+  const formControl = input.parentElement;
+  formControl.className = 'form-control success';
+}
+
+function isEmail(email) {
+  return email === email.toLowerCase();
+}
+
+function checkInputs() {
+  let valid = true;
+  let errorMsg = '';
+
+  const emailValue = email.value.trim();
+  if (emailValue === '') {
+    valid = false;
+    errorMsg = 'Email cannot be blank';
+    setErrorFor(email, errorMsg);
+  } else if (!isEmail(emailValue)) {
+    valid = false;
+    errorMsg = 'Email should be in lowercase';
+    setErrorFor(email, errorMsg);
+  } else {
+    setSuccessFor(email);
+  }
+
```

Comment:
- [ OPTIONAL ] I think it would be great if you style the error message since I just see it is plain but it would be great if you style it as well ⭐✅✨